roles/openshift_control_plane/templates/master.yaml.v1.j2

```diff
@@ -77,16 +77,26 @@ kubernetesMasterConfig:
     - etcd3
     storage-media-type:
     - application/vnd.kubernetes.protobuf
-{% if openshift_master_use_persistentlocalvolumes | bool %}
+{% if (openshift_master_use_persistentlocalvolumes | bool) or (openshift_master_use_ephemeralstorage | bool) %}
```

Comment:
Lets refactor this - add a new `l_openshift_feature_gates` var, which would contain a list of feature gates to enable

Reply:
Agree, this is a prototype but generalizing the feature gates would be much better.

Reply:
There turn out to be three different sets of feature gates used in different contexts.  There are also three different conditions used at present (see below).

There has been discussion (and some PRs) in kubernetes/kubernetes about the subject of different feature gates for different components.  While https://github.com/kubernetes/kubernetes/pull/55067 was not merged, there appeared to be broad consensus that all component should accept all feature gates even if they don't do anything with them.  The documentation at https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ does not describe situations wherein it would be desirable or necessary to have different feature gate settings on different components.

If that is in fact the situation, we could compose a list of feature gates at a high level that is used in all places where feature gates can be set.

```roles/openshift_control_plane/templates/master.yaml.v1.j2:
roles/openshift_node/templates/node.yaml.v1.j2:

* PersistentLocalVolumes
* VolumeScheduling
+ LocalStorageCapacityIsolation

roles/openshift_node/tasks/upgrade/bootstrap_changes.yml:
roles/openshift_node_group/tasks/upgrade.yml:
roles/openshift_node_group/templates/node-config.yaml.j2:

. RotateKubeletClientCertificate
. RotateKubeletServerCertificate
+ LocalStorageCapacityIsolation

roles/openshift_service_catalog/templates/controller_manager.j2:
. OriginatingIdentity=true
= AsyncBindingOperations
+ LocalStorageCapacityIsolation

Legend:
. unconditional
* conditional on use_persistentlocalvolumes
+ conditional on use_ephemeralstorage
= conditional on async_bindings_enabled```

Reply:
Testing has determined that not all feature gates can be used with all components.  In particular, the API server does not simply ignore LocalStorageCapacityIsolation; it actively throws an error.

So we can't simply have a list of feature gates to be added, then add them everywhere that feature gates are specified.  We need to know which feature gates apply to which components.